deployment/src/main/java/org/wso2/testgrid/deployment/DeploymentUtil.java

```diff
@@ -38,21 +43,36 @@
     /**
      * Reads the deployment.json file and constructs the deployment object.
      *
-     * @param testPlanLocation location String of the test plan
+     * @param workspace location String of the test plan
      * @return the deployment information ObjectMapper
      * @throws TestGridDeployerException If reading the deployment.json file fails
      */
-    public static DeploymentCreationResult getDeploymentCreationResult(String testPlanLocation)
+    public static DeploymentCreationResult getDeploymentCreationResult(String workspace)
             throws TestGridDeployerException {
-
+        DeploymentCreationResult deploymentCreationResult = new DeploymentCreationResult();
+        List<Host> hosts = new ArrayList<>();
         ObjectMapper mapper = new ObjectMapper();
-        File file = new File(Paths.get(testPlanLocation, DeployerConstants.DEPLOYMENT_FILE).toString());
+        File file = new File(Paths.get(workspace, DeployerConstants.DEPLOYMENT_FILE).toString());
         try {
-            return mapper.readValue(file, DeploymentCreationResult.class);
+            List<Host> hostList = mapper.readValue(file, DeploymentCreationResult.class).getHosts();
+            for (Host host : hostList) {
```

Comment:
IMO we can simplify this with a lambda statement. WDYT?

Reply:
@vidurananayakkara : IMO this code won't gain anything new if we use a lambda. Is there any particular reason behind your suggestion?

Reply:
We have 2 loops inside a try. I was thinking about the possibility of reducing that to few lines which will make the code more readable IMO as well. For the moment lets proceed with what we have.